processing/src/main/java/io/druid/query/groupby/epinephelinae/RowBasedGrouperHelper.java

```diff
@@ -172,10 +179,27 @@
           if (isInputRaw) {
             IndexedInts index = dimensionSelectors[i - dimStart].getRow();
             value = index.size() == 0 ? "" : dimensionSelectors[i - dimStart].lookupName(index.get(0));
+            key[i] = Strings.nullToEmpty(value);
           } else {
-            value = (String) row.getRaw(query.getDimensions().get(i - dimStart).getOutputName());
+            // only GroupByV2 (isInputRaw == false) supports non-String types
```

Comment:
Adjusted this area to create a value supplier for each column based on the row signature, when isInputRaw is true, and added a comment to the method on isInputRaw